packages/storage/amplify_storage_s3_dart/test/storage_s3_service/task/s3_upload_task_test.dart

```diff
@@ -139,6 +140,58 @@ void main() {
         expect(request.body, testDataPayload);
       });
 
+      test(
+        'should use fallback contentType header when contentType of the data'
+        ' payload is not determinable',
+        () async {
+          const testUploadDataOptions = S3UploadDataOptions(
+            accessLevel: StorageAccessLevel.private,
+          );
+          final testPutObjectOutput = s3.PutObjectOutput();
+          final smithyOperation = MockSmithyOperation<s3.PutObjectOutput>();
+
+          when(
+            () => smithyOperation.result,
+          ).thenAnswer((_) async => testPutObjectOutput);
+          when(() => smithyOperation.requestProgress)
+              .thenAnswer((_) => Stream.value(1));
+
+          when(
+            () => s3Client.putObject(any()),
+          ).thenAnswer((_) => smithyOperation);
+
+          final uploadDataTask = S3UploadTask.fromDataPayload(
+            testDataPayloadBytes,
+            s3Client: s3Client,
+            prefixResolver: testPrefixResolver,
+            bucket: testBucket,
+            key: testKey,
+            options: testUploadDataOptions,
+            logger: logger,
+            transferDatabase: transferDatabase,
+          );
+
+          unawaited(uploadDataTask.start());
+
+          final result = await uploadDataTask.result;
+
+          expect(result.key, testKey);
```

Comment:
nit/question: Is it necessary to repeat the assertions from previous test when this test is checking something totally different (for which it has a new assertion which is great)?

Reply:
Ahh yea this assertion is not needed here, could be removed. 
This test focuses on the fallback `contentType` value.